### DISCUSSION TITLE
Too much time for kkt command

### DISCUSSION THREAD

**Discussion by rgzbsn** [yalmip/YALMIP#1280] [Problems using YALMIP] [answered]
Hi Johan and everyone
 I need to create a KKT condition of a LP model,and thus calling the kkt command.But it do take a lot of time(more than 15m) to complete the kkt command.I'm wondering how to promote the efficiency of the command.
Here is some message of the model:
```
Removed 217 duplicated or redundant inequalities
Transfered 2534 inequalities to equalities
Starting derivation of dual bounds (can be turned off using option kkt.dualbounds)
*Computing 1992 primal bounds (required for dual bounds)
*Computing 4905 bounds on parameterized RHS (required for dual bounds)
*Computing 4905 dual bounds
```
and codes related kkt command are as follow:
```
[KKTsystem,details]=kkt(Cint,Obj,u);
ops=sdpsettings('solver','gurobi','gurobi.MIPGap',0.005);
optimize([KKTsystem,Cint,details.dual<=100],-Obj,ops);
```
Thanks in advance!

**Comment by johanlofberg** [comment, marked as answer]
As it says, turn off  kkt.dualbounds

However, you are probably going to struggle with this model. It is way beyond intended size, and now that you turn off computation of good bounds, the big-M model will likely be horrible numerically (if it wasn't before), and you will have to manually add bounds on the duals (details.duals) and then you probably have the problem that you have no idea what reasonable bounds are, thus either using too tight bounds and risk throwing away the optimal solution, or using too large bounds and causing numerical disasters

**Reply by rgzbsn** [reply]
Thank you for your response johan!But I don't know how to turn off kkt.dualbounds(say,set kkt.dualbounds to 0 or something?)

**Reply by johanlofberg** [reply]
https://yalmip.github.io/command/sdpsettings/

**Reply by rgzbsn** [reply]
Thank you!

**Reply by ZihanXu1-debug** [reply]
Have you figured out how to turn it off？

**Reply by johanlofberg** [reply]
As been said already, you set kkt.dualbounds to 0 in sdpsettings

**Reply by ZihanXu1-debug** [reply]
Thank you

**Comment by ZihanXu1-debug** [comment]
![3ZSG~0%E}RC@R7ZFI BG5WD](https://github.com/yalmip/YALMIP/assets/174466230/f2745b3f-3e0b-4c40-a1df-4d54958519f8)
Sorry to bother you.Why does it still show that it is not closed after I fill it in?

**Reply by johanlofberg** [reply]
?? You create the options *after* you make the call, and you don't use the options in the call (of course, as you hadn't even defined it)
